DifferenceGenerator/src/main/kotlin/DifferenceGenerator.kt

```diff
@@ -140,8 +164,23 @@ class DifferenceGenerator(video1Path: String, video2Path: String, outputPath: St
      * @param color the color
      * @return a frame colored in the given color
      */
-    private fun getFrame(color: Color): Frame {
+    private fun getColoredFrame(color: Color): Frame {
         val converterOutput = Java2DFrameConverter()
```

Comment:
a new Java2DFrameConverter is declared in almost every function of this class. Maybe it would be better memory-wise to declare a global one and just use it in the different methods?

Reply:
This also applies to Graphics2D

Reply:
This was already discussed in another PR, where there was seemingly no option to use one converter for two frames, as the second conversion would be influenced by the first one without an option to reset the converter
Let me find the discussion

Reply:
https://github.com/amosproj/amos2023ws03-gui-frame-diff/pull/42#issuecomment-1808075689

Reply:
Other than that I agree with you, that this seems like a bad practice, but I dont think that JavaCV lets us do it much differently at this point

Reply:
Ah yes I didn't read it the first time. That's kinda unfortunate but apparently necessary somehow.